python/cuml/testing/strategies.py

```diff
@@ -521,3 +531,92 @@ def standard_regression_datasets(
     strategies.
     """
 )
+
+
+@composite
+def standard_classification_datasets(
+    draw,
+    dtypes=floating_dtypes(),
+    n_samples=integers(min_value=100, max_value=200),
+    n_features=integers(min_value=10, max_value=20),
+    *,
+    n_informative=None,
+    n_redundant=None,
+    n_repeated=just(0),
+    n_classes=just(2),
+    n_clusters_per_class=just(2),
+    weights=none(),
+    flip_y=just(0.01),
+    class_sep=just(1.0),
+    hypercube=just(True),
+    shift=just(0.0),
+    scale=just(1.0),
+    shuffle=just(True),
+    random_state=None,
+    labels_dtypes=integer_dtypes(),
+):
+    n_features_ = draw(n_features)
+    if n_informative is None:
+        try:
+            # Try to meet:
+            #   log_2(n_classes * n_clusters_per_class) <= n_informative
+            n_classes_min = min(_get_limits(n_classes))
+            n_clusters_per_class_min = min(_get_limits(n_clusters_per_class))
+            n_informative_min = \
+                int(np.ceil(np.log2(n_classes_min * n_clusters_per_class_min)))
+        except AttributeError:
+            # Otherwise aim for 10% of n_features, but at least 1.
+            n_informative_min = max(1, int(0.1 * n_features_))
+
+        n_informative = just(min(n_features_, n_informative_min))
+    if n_redundant is None:
+        n_redundant = just(max(min(n_features_, 1), int(0.1 * n_features_)))
+
+    # Check whether the
+    #   log_2(n_classes * n_clusters_per_class) <= n_informative
+    # inequality can in principle be met.
+    try:
+        n_classes_min = min(_get_limits(n_classes))
+        n_clusters_per_class_min = min(_get_limits(n_clusters_per_class))
+        n_informative_max = max(_get_limits(n_informative))
+    except AttributeError:
+        pass  # unable to determine limits
+    else:
+        if np.log2(n_classes_min * n_clusters_per_class_min) \
+                > n_informative_max:
+            raise ValueError(
+                "Assumptions cannot be met, the following inequality must "
+                "hold: log_2(n_classes * n_clusters_per_class) "
+                "<= n_informative ."
+            )
+
+    # Check base assumption concerning the composition of feature vectors.
+    n_informative_ = draw(n_informative)
+    n_redundant_ = draw(n_redundant)
+    n_repeated_ = draw(n_repeated)
+    assume(n_informative_ + n_redundant_ + n_repeated_ < n_features_)
```

Comment:
A `<=` should be safe here, right? Is there another category of features I'm not thinking of?

Reply:
Yes, should be fine indeed.